server/src/internalClusterTest/java/org/elasticsearch/snapshots/DataStreamsSnapshotsIT.java

```diff
@@ -0,0 +1,224 @@
+        assertEquals(1, ds.getDataStreams().get(0).getIndices().size());
+        assertEquals(".ds-ds2-000001", ds.getDataStreams().get(0).getIndices().get(0).getName());
+    }
+
+    public void testDataStreamNotStoredWhenIndexRequested() throws Exception {
+        Client client = client();
+
+        Path location = randomRepoPath();
+        createRepository("repo", "fs", location);
+
+        DataStreamIT.createIndexTemplate("t1", "@timestamp", "ds", "other-ds");
+
+        CreateDataStreamAction.Request request = new CreateDataStreamAction.Request("ds");
+        AcknowledgedResponse response = client.admin().indices().createDataStream(request).get();
+        assertTrue(response.isAcknowledged());
+
+        CreateSnapshotResponse createSnapshotResponse = client.admin().cluster()
+            .prepareCreateSnapshot("repo", "snap2")
+            .setWaitForCompletion(true)
+            .setIndices(".ds-ds-000001")
+            .setIncludeGlobalState(false)
+            .get();
+
+        RestStatus status = createSnapshotResponse.getSnapshotInfo().status();
+        assertEquals(RestStatus.OK, status);
+        expectThrows(Exception.class, () -> client.admin().cluster()
+            .prepareRestoreSnapshot("repo", "snap2")
+            .setWaitForCompletion(true)
+            .setIndices("ds")
+            .get());
+    }
+
+    public void testDataStreamNotRestoredWhenIndexRequested() throws Exception {
+        Client client = client();
+
+        Path location = randomRepoPath();
+        createRepository("repo", "fs", location);
+
+        DataStreamIT.createIndexTemplate("t1", "@timestamp", "ds", "other-ds");
+
+        CreateDataStreamAction.Request request = new CreateDataStreamAction.Request("ds");
+        AcknowledgedResponse response = client.admin().indices().createDataStream(request).get();
+        assertTrue(response.isAcknowledged());
+
+        CreateSnapshotResponse createSnapshotResponse = client.admin().cluster()
+            .prepareCreateSnapshot("repo", "snap2")
+            .setWaitForCompletion(true)
+            .setIndices("ds")
+            .setIncludeGlobalState(false)
+            .get();
+
+        RestStatus status = createSnapshotResponse.getSnapshotInfo().status();
+        assertEquals(RestStatus.OK, status);
+
+        assertTrue(client.admin().indices().deleteDataStream(new DeleteDataStreamAction.Request("ds")).get().isAcknowledged());
+
+        RestoreSnapshotResponse restoreSnapshotResponse = client.admin().cluster()
+            .prepareRestoreSnapshot("repo", "snap2")
+            .setWaitForCompletion(true)
+            .setIndices(".ds-ds-*")
+            .get();
+
+        assertEquals(RestStatus.OK, restoreSnapshotResponse.status());
+
+        GetDataStreamAction.Request getRequest = new GetDataStreamAction.Request("ds");
+        Throwable e = expectThrows(ExecutionException.class, () -> client.admin().indices().getDataStreams(getRequest).get()).getCause();
```

Comment:
iirc if `actionGet()` is invoked then the real exception is thrown instead of `ExecutionException`. So maybe do that instead of invoked `get()`?

Reply:
Great suggestion, thanks!